### PR TITLE
#139: Added PayloadHandler

### LINKED ISSUE #139: Add PayloadHandler as Facade for PayloadSender and PayloadReceiver

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #139

### CODE REVIEW

File: exasol_advanced_analytics_framework/udf_communication/peer_communicator/payload_handler.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def received_acknowledge_payload(self, message: messages.AcknowledgePayload):
+        self._payload_sender.received_acknowledge_payload(message)
+
+    def received_payload(self, message: messages.Payload, frames: List[Frame]):
+        self._payload_receiver.received_payload(message, frames)
+
+    def try_send(self):
+        self._payload_sender.try_send()
+
+    def is_ready_to_stop(self):
```
Comment (tomuben) [inline exasol_advanced_analytics_framework/udf_communication/peer_communicator/payload_handler.py:28]: add type hint for return value

File: tests/unit_tests/udf_communication/peer_communication/test_payload_handler.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+def test_init():
+    test_setup = create_test_setup()
+    assert test_setup.payload_receiver_mock.mock_calls == [] and \
+           test_setup.payload_sender_mock.mock_calls == []
+
+
+def test_try_send():
+    test_setup = create_test_setup()
```
Comment (tomuben) [inline tests/unit_tests/udf_communication/peer_communication/test_payload_handler.py:48]: does it make sense to create a fixture for that like:
```
@pytest.fixture
def test_setup():
    test_setup = create_test_setup()
    test_setup.reset_mock()
    return test_setup
```

Comment (tkilias) [reply tests/unit_tests/udf_communication/peer_communication/test_payload_handler.py:48]: puh, good question. we need the reset_mock in every test except `test_init`, but on the other side it is only function call per test. Maybe more like extract a function for it.

File: tests/unit_tests/udf_communication/peer_communication/test_payload_handler.py [hunk 3]
```diff
@@ -45,16 +45,19 @@ def test_init():
...
     test_setup.payload_handler.try_send()
     assert test_setup.payload_receiver_mock.mock_calls == [] and \
            test_setup.payload_sender_mock.mock_calls == [call.try_send()]
 
 
-def test_send_payload():
+def create_resetted_test_setup():
     test_setup = create_test_setup()
-    test_setup.reset_mock()
+    return test_setup
```
Comment (tomuben) [inline tests/unit_tests/udf_communication/peer_communication/test_payload_handler.py:56]: Missing:
```
 test_setup.reset_mock()
```

**Review by tomuben — APPROVED** [review]

**Pull request merged**